pipelines/generateServiceLibrary.yml

```diff
@@ -0,0 +1,189 @@
+    source: (beta - 3) msgraph-publish-cleanmetadata
+    trigger: # TODO: validate that this pipeline triggers as expected.
+      branches:
+      - master
+
+pool:
+  vmImage: windows-latest # Info about this image: [0][1]
+
+variables:
+  - group: MicrosoftGraph # Variable group, where variables not set here come from. Set in Azure DevOps
+  - name: endpoint
+    value: beta
+
+steps:
+- checkout: msgraph-beta-sdk-dotnet
+  clean: true
+  fetchDepth: 1
+  #persistCredentials: true
+- checkout: msgraph-metadata
+  clean: true
+  fetchDepth: 1
+
+- task: PowerShell@2 # Setup environment variables and make them available to all tasks. See [1] for more info.
+  displayName: 'Calculate and set pipeline variables for this job'
+  inputs:
+    targetType: inline
+    script: |
+      
+      $repoModelsDir = "$env:Build_SourcesDirectory\msgraph-beta-sdk-dotnet\src\Microsoft.Graph\Models\Generated\"
+      $repoRequestDir = "$env:Build_SourcesDirectory\msgraph-beta-sdk-dotnet\src\Microsoft.Graph\Requests\Generated\"
+      Write-Host "Path to repo model directory: $repoModelsDir"
+      Write-Host "Path to repo model directory: $repoRequestDir"
+      Write-Host "##vso[task.setvariable variable=repoModelsDir]$repoModelsDir"
+      Write-Host "##vso[task.setvariable variable=repoRequestDir]$repoRequestDir"
+
+      $outputPath = Join-Path $env:Build_SourcesDirectory "output"
+      Write-Host "Path to typewriter.exe output $outputPath"
+      Write-Host "##vso[task.setvariable variable=outputPath]$outputPath"
+
+      $cleanMetadata = "https://raw.githubusercontent.com/microsoftgraph/msgraph-metadata/master/clean_beta_metadata/cleanMetadataWithDescriptionsbeta.xml"
+      Write-Host "Path to clean metadata $cleanMetadata"
+      Write-Host "##vso[task.setvariable variable=cleanMetadata]$cleanMetadata"
+
+      $branchName = "beta/pipelinebuild/$env:Build_BuildId" # Match the spec in the GH Action
+      Write-Host "Branch path spec for the pull request will be $branchName"
+      Write-Host "##vso[task.setvariable variable=branchName]$branchName"
+
+- task: PowerShell@2
+  displayName: 'Git: checkout master'
+  inputs:
+    targetType: inline
+    workingDirectory: '$(Build.SourcesDirectory)/msgraph-beta-sdk-dotnet'
+    script: |
+      git checkout master | Write-Host
+
+- task: PowerShell@2
+  displayName: 'Git: branch from master named with the build id: $(Build.BuildId)'
+  inputs:
+    targetType: inline
+    workingDirectory: '$(Build.SourcesDirectory)/msgraph-beta-sdk-dotnet'
+    script: |
+      
+      Write-Host "The new branch name will be: $env:branchName"
+      git checkout -B $env:branchName | Write-Host
+
+- task: PowerShell@2
+  displayName: 'Git: set user config'
+  inputs:
+    targetType: inline
+    workingDirectory: '$(Build.SourcesDirectory)/msgraph-beta-sdk-dotnet'
+    script: |
+      git config user.email "GraphTooling@service.microsoft.com"
```

Comment:
Is `git config` also affected by the inline script issue? We are making two git commands here.

Reply:
It isn't. It seemed like it was an issue when working with branches.